libs/@blockprotocol/type-system/rust/src/schema/data_type/constraint/number.rs

```diff
@@ -549,4 +779,651 @@ mod tests {
+                    .expect("Failed to parse schema"),
+                ),
+                ResolveClosedDataTypeError::UnsatisfiedEnumConstraintVariant(json!(5.0)),
+                ResolveClosedDataTypeError::UnsatisfiedEnumConstraintVariant(json!(10.0)),
+                ResolveClosedDataTypeError::UnsatisfiedEnumConstraintVariant(json!(15.0)),
+            ],
+        );
+    }
+
+    #[test]
+    fn intersect_mixed() {
+        check_schema_intersection(
+            [
+                json!({
+                    "type": "number",
+                    "minimum": 5.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 5.0,
+                }),
+                json!({
+                    "type": "number",
+                    "maximum": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 3.0,
+                }),
+                json!({
+                    "type": "number",
+                    "maximum": 15.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "minimum": 7.0,
+                    "maximum": 20.0,
+                }),
+            ],
+            [
+                json!({
+                    "type": "number",
+                    "minimum": 7.0,
+                    "maximum": 10.0,
+                    "multipleOf": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 3.0,
+                }),
+            ],
+        );
+
+        check_schema_intersection(
+            [
+                json!({
+                    "type": "number",
+                    "minimum": 5.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 5.0,
+                }),
+                json!({
+                    "type": "number",
+                    "maximum": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "maximum": 15.0,
+                }),
+                json!({
+                    "type": "number",
+                    "const": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "minimum": 10.0,
+                    "maximum": 10.0,
+                }),
+                json!({
+                    "type": "number",
+                    "multipleOf": 2.0,
+                }),
+            ],
+            [json!({
+                "type": "number",
+                "const": 10.0,
+            })],
+        );
+    }
```

Comment:
non-blocking, but I am guessing schema validation stops `minimum: 7, maximum: 5` from occurring on the same object?